Remove "canonical" adjective from Status Code (#1081)
open-telemetry/opentelemetry-specification: specification/trace/api.md

`Status` is semantically defined by the following properties:

> - `Code`, one of the values listed below.

Reviewer: Despite being redundant, I would not be opposed to calling it `StatusCode`.
This would allow for statements like "`StatusCode` has to be set to ..." rather than "`Code` of/on `Status` "or "`Status`' `Code`".

Reviewer 2: +1 - I suspect most languages will name it something similar anyways. And the PR title also says remove canonical but keeps Status Code :)

Reviewer 3: +1 I think calling it `StatusCode` in the specs is better.

Reviewer: ```suggestion
- `StatusCode`, one of the values listed below.
```

Author: Done.

- Optional `Description` that provides a descriptive message of the `Status`.

> `Code` is one of the following values:

Reviewer: ```suggestion
`StatusCode` is one of the following values:
```

Author: Done.

The Span interface MUST provide:

> - An API to set the `Status`. This SHOULD be called `SetStatus`. This API takes

Reviewer 4: Preferably keep the linebreak at sentence end.

Author: Fixed.

>   the `Code`, and optional `Description`, either as individual parameters or as

Reviewer 4: ```suggestion
  the `Status`' `Code`, and optional `Description`, either as individual parameters or as
```

Author: It looks a bit less readable, so I'd prefer not to. I think in this context it is clear that `Code` refers to the Status's code and not something else.

Reviewer 4: Fine with me.

The Span interface MUST provide:

- An API to set the `Status`. This SHOULD be called `SetStatus`.

>   This API takes the `Code`, and optional `Description`, either as individual

Reviewer: ```suggestion
  This API takes the `Code`, and an optional `Description`, either as individual
```

Reviewer: ```suggestion
  This API takes the `StatusCode`, and an optional `Description`, either as individual
```

Author: Done.
